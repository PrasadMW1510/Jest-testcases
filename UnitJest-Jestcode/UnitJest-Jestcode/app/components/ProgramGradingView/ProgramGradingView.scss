.programgrading-header {
  padding-top: 5px;
}

.programgrading-header__name {
  float: left;
  font-weight: bold;
  font-size: 14px;
}

.programgrading-header__image {
  float: right;
}

.programgrading-header__fm {
  padding-top: 5px;

  .programgrading-header__image {
    width: 140px;
    height: 37px;
  }
}

.programgrading-header__src {
  padding-top: 5px;

  .programgrading-header__name {
    width: 260px;
    line-height: 18px;
    font-size: 16px;
  }
}
