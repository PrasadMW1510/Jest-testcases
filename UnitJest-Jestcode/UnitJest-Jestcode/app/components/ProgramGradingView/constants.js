export const UNDER_CONSTRUCTION = 'The Grading Tools is Under Construction';

export const TAB_SETTINGS = {
  id: 'tab-settings',
  label: 'Settings',
};

export const APPLICATIONS_IDS = {
  dtm: ['DTM'],
  e3d: ['E3D'],
  fm: ['FM'],
  fmng: ['FMNG'],
  r180: ['R180'],
  r180u: ['R180U'],
  RT: ['RT'],
  RTNG: ['RTNG'],
  s44: ['S44'],
  s44jr: ['S44JR'],
  smi: ['SMI'],
  src: ['SRC'],
  sri: ['SRI'],
  xt: ['XT'],
};
