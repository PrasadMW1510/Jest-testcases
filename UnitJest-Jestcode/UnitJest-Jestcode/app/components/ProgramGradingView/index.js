/**
 *
 * ProgramGradingView
 *
 */

export { default } from './ProgramGradingView';
