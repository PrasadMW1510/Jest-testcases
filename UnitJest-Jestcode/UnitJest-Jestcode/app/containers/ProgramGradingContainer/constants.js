/*
 *
 * ProgramGradingContainer constants
 *
 */

export const PROGRAM_GRADING_ENROLLMENT_LIST =
  'app/ProgramGradingContainer/PROGRAM_GRADING_ENROLLMENT_LIST';
export const PROGRAM_GRADING_ENROLLMENT_LIST_SUCCESS =
  'app/ProgramGradingContainer/PROGRAM_GRADING_ENROLLMENT_LIST_SUCCESS';
export const PROGRAM_GRADING_ENROLLMENT_LIST_FAILURE =
  'app/ProgramGradingContainer/PROGRAM_GRADING_ENROLLMENT_LIST_FAILURE';
