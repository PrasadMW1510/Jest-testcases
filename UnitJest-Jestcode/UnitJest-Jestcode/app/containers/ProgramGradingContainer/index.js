/**
 *
 * ProgramGradingContainer
 *
 */

export { default } from './ProgramGradingContainer';
