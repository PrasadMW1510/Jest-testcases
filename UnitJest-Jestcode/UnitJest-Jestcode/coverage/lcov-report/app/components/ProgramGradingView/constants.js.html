<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/ProgramGradingView/constants.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">app/components/ProgramGradingView</a> constants.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/3</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const <span class="cstat-no" title="statement not covered" >UNDER_CONSTRUCTION = 'The Grading Tools is Under Construction';<span class="cstat-no" title="statement not covered" ></span></span>
&nbsp;
export const TAB_SE<span class="cstat-no" title="statement not covered" >TTINGS = {</span>
  id: 'tab-settings',
  label: 'Settings',
};
&nbsp;
export const APPLICATIO<span class="cstat-no" title="statement not covered" >NS_IDS = {</span>
  dtm: ['DTM'],
  e3d: ['E3D'],
  fm: ['FM'],
  fmng: ['FMNG'],
  r180: ['R180'],
  r180u: ['R180U'],
  RT: ['RT'],
  RTNG: ['RTNG'],
  s44: ['S44'],
  s44jr: ['S44JR'],
  smi: ['SMI'],
  src: ['SRC'],
  sri: ['SRI'],
  xt: ['XT'],
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 21 2018 17:06:47 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
