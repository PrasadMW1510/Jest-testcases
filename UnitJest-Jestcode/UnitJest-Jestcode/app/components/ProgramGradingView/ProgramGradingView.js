/**
 *
 * ProgramGradingView
 *
 */

import React from 'react';

import PropTypes from 'prop-types';
import { PROGRAM_LIST } from 'containers/App/constants';
import { getEnrollmentCount } from 'utils/programSettingsUtils';
import SettingsMessage from 'components/SettingsMessage';
import ProgramSettingsNavBar from 'components/ProgramSettingsNavBar';
import SRCGradingToolsContainer from 'containers/SRCGradingToolsContainer/Loadable';
import FMGradingToolsContainer from 'containers/FMGradingToolsContainer/Loadable';

import * as Constants from './constants';
import './ProgramGradingView.scss';

class ProgramGradingView extends React.Component {
  constructor(props) {
    super(props);

    this.headerName = '';
    this.productClassName = 'programgrading-header';
  }

  renderProgramSettings = () => {
    const { enrollmentList } = this.props;
    // Keep these cases in alphabetical order
    switch (this.props.selectedProgram.product_code) {
      // case PROGRAM_LIST.DTM.code: {
      //   const fadEnrollmentCount = this.getEnrollmentCount(Constants.APPLICATIONS_IDS.dtm);
      //   return <FADSettingsContainer enrollmentCount={fadEnrollmentCount} />;
      // }
      case PROGRAM_LIST.FM.code: {
        this.headerName = 'Worksheets';
        this.productClassName = 'programgrading-header__fm';
        const fmEnrollmentCount = getEnrollmentCount(Constants.APPLICATIONS_IDS.fm, enrollmentList);
        return <FMGradingToolsContainer enrollmentCount={fmEnrollmentCount} />;
      }
      // case PROGRAM_LIST.R180NG.code:
      //   return <R180NGSettingContainer />;
      // case PROGRAM_LIST.RTNG.code: {
      //   const rSkillsCCEnrollmentList = this.getEnrollmentCount(Constants.APPLICATIONS_IDS.rtng);
      //   return <RSkillsCCSettingContainer enrollmentCount={rSkillsCCEnrollmentList} />;
      // }
      // case PROGRAM_LIST.S44.code: {
      //   const s44EnrollmentCount = this.getEnrollmentCount(Constants.APPLICATIONS_IDS.s44);
      //   return <S44SettingContainer enrollmentCount={s44EnrollmentCount} />;
      // }
      // case PROGRAM_LIST.S44JR.code: {
      //   const s44JREnrollmentCount = this.getEnrollmentCount(Constants.APPLICATIONS_IDS.s44jr);
      //   return <IreadSettingsContainer enrollmentCount={s44JREnrollmentCount} />;
      // }
      // case PROGRAM_LIST.S44NG.code: {
      //   const s44NGEnrollmentCount = this.getEnrollmentCount(Constants.APPLICATIONS_IDS.s44ng);
      //   return <S44NGSettingContainer enrollmentCount={s44NGEnrollmentCount} />;
      // }
      // case PROGRAM_LIST.SMI.code: {
      //   const smiEnrollmentCount = this.getEnrollmentCount(Constants.APPLICATIONS_IDS.smi);
      //   return <MISettingContainer enrollmentCount={smiEnrollmentCount} />;
      // }
      // case PROGRAM_LIST.SRI.code: {
      //   const sriEnrollmentCount = this.getEnrollmentCount(Constants.APPLICATIONS_IDS.sri);
      //   return <RISettingContainer enrollmentCount={sriEnrollmentCount} />;
      // }
      case PROGRAM_LIST.SRC.code: {
        this.productClassName = 'programgrading-header__src';
        this.headerName = 'Points Usage and Quiz Score Recording Tools';
        const srcEnrollmentCount = getEnrollmentCount(
          Constants.APPLICATIONS_IDS.src,
          enrollmentList
        );
        return <SRCGradingToolsContainer enrollmentCount={srcEnrollmentCount} />;
      }
      default: {
        const tabs = [
          {
            renderFunction: () => <SettingsMessage message1={Constants.UNDER_CONSTRUCTION} />,
            ...Constants.TAB_SETTINGS,
          },
        ];

        return <ProgramSettingsNavBar tabs={tabs} />;
      }
    }
  };
  render() {
    const { selectedProgram } = this.props;
    return (
      <div>
        <div className={this.productClassName}>
          <span className="programgrading-header__name">
            {selectedProgram.display_name} {` ${this.headerName}`}
          </span>
          <img
            className="programgrading-header__image"
            src={selectedProgram.display_image_large}
            alt={selectedProgram.display_name}
          />
          {this.renderProgramSettings()}
        </div>
      </div>
    );
  }
}

ProgramGradingView.propTypes = {
  enrollmentList: PropTypes.object.isRequired,
  selectedProgram: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
};

export default ProgramGradingView;
