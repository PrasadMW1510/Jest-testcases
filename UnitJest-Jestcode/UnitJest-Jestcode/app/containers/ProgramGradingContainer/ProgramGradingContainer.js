/**
 *
 * ProgramGradingContainer
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

// (see note below about imports from ProgramSettingsContainer)
import ProgramGradingView from 'components/ProgramGradingView';
import { programSettingsEnrollmentList } from 'containers/ProgramSettingsContainer/actions';
import reducer from 'containers/ProgramSettingsContainer/reducer';
import saga from 'containers/ProgramSettingsContainer/saga';
import { makeSelectEnrollmentList } from 'containers/ProgramSettingsContainer/selectors';
import makeSelectRosterPage from 'containers/RosterPage/selectors';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

/**
 * NOTE:  The ProgramGradingContainer is a peer of the Program Settings
 * and reuses a lot of the same data retrieval methods, e.g.
 * actions and selectors.  We opt to use the Program Settings as the
 * primary controller of such info (rather than Grading Tools).
 */
export class ProgramGradingContainer extends React.Component {
  componentDidMount() {
    this.props.programSettingsEnrollmentList();
  }
  render() {
    return (
      <div className="roster-content-panel">
        <ProgramGradingView
          selectedProgram={this.props.rosterPage.selectedProgram}
          enrollmentList={this.props.enrollmentList}
        />
      </div>
    );
  }
}

ProgramGradingContainer.propTypes = {
  enrollmentList: PropTypes.object.isRequired,
  programSettingsEnrollmentList: PropTypes.func.isRequired,
  rosterPage: PropTypes.object.isRequired,
};

const mapStateToProps = createStructuredSelector({
  enrollmentList: makeSelectEnrollmentList(),
  rosterPage: makeSelectRosterPage(),
});

const withConnect = connect(mapStateToProps, { programSettingsEnrollmentList });
const withReducer = injectReducer({ key: 'programSettingsContainer', reducer });
const withSaga = injectSaga({ key: 'programSettingsContainer', saga });

export default compose(withReducer, withSaga, withConnect)(ProgramGradingContainer);
