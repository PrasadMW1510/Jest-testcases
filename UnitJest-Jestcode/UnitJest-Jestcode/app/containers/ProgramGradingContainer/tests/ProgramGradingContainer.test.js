import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { fromJS } from 'immutable';

import ProgramGradingView from 'components/ProgramGradingView';
import { ProgramGradingContainer } from '../ProgramGradingContainer';

describe('<ProgramGradingContainer />', () => {
  let wrapper = null;
  let mockProgramSettingsEnrollmentList = null;

  const mockEnrollmentList = fromJS([]);
  const mockRosterPage = {
    selectedProgram: {
      settings: 'Settings',
      display_name: 'Progress Space',
      display_image: '/a2dabdfecfde82f90145cbd702e07a6b.png',
    },
  };

  beforeEach(() => {
    mockProgramSettingsEnrollmentList = jest.fn();

    wrapper = shallow(
      <ProgramGradingContainer
        enrollmentList={mockEnrollmentList}
        programSettingsEnrollmentList={mockProgramSettingsEnrollmentList}
        rosterPage={mockRosterPage}
      />
    );
  });

  it('Expect to render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should contain a ProgramGradingView', () => {
    expect(wrapper.find(ProgramGradingView).exists()).toBeTruthy();
  });
});
