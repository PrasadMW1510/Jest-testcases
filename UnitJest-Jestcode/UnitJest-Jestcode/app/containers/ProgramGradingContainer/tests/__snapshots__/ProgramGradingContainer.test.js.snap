// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgramGradingContainer /> Expect to render correctly 1`] = `
<div
  className="roster-content-panel"
>
  <ProgramGradingView
    enrollmentList={
      Immutable.List [
      ]
    }
    selectedProgram={
      Object {
        "display_image": "/a2dabdfecfde82f90145cbd702e07a6b.png",
        "display_name": "Progress Space",
        "settings": "Settings",
      }
    }
  />
</div>
`;
